src/rgw/rgw_data_sync.cc

```diff
@@ -763,21 +784,33 @@ class RGWListBucketIndexesCR : public RGWCoroutine {
 
   int operate() override {
     reenter(this) {
+      entries_index = new RGWShardedOmapCRManager(sync_env->async_rados, store, this, num_shards,
+						  store->svc.zone->get_zone_params().log_pool,
+                                                  oid_prefix);
+      yield; // yield so OmapAppendCRs can start
+
+      do {
       yield {
+
+        char max_entries_buf[32];
+        snprintf(max_entries_buf, sizeof(max_entries_buf), "%d", (int)max_entries);
+        
         string entrypoint = string("/admin/metadata/bucket.instance");
-        /* FIXME: need a better scaling solution here, requires streaming output */
-        call(new RGWReadRESTResourceCR<list<string> >(store->ctx(), sync_env->conn, sync_env->http_manager,
-                                                      entrypoint, NULL, &result));
+
+        rgw_http_param_pair pairs[] = { { "type", "data" },
+        { "max-entries", max_entries_buf },
```

Comment:
can remove the `type=data` param, it isn't used by the /admin/metadata/ api (RGWOp_Metadata_List)

for max-entries, we might as well hardcode "1000" here to avoid the GET_MAX_ENTRIES define, the max_entries variable, and the max_entries_buf/snprintf. 1000 was the existing default from RGWOp_Metadata_List, and i think that's still reasonable